compile.php

```diff
@@ -28,12 +28,47 @@ function strip_decl($str)
 foreach($refs as $ref)
 {
     $ref = trim($ref);
-    $safedoi = str_replace(')', '_', str_replace('(', '_', str_replace('/', '_', $ref)));
-    $outfile = "tmp/ref_$safedoi.xml";
-    if(!file_exists($outfile))
+    if(strpos($ref, 'doi:') === 0)
     {
-        exec("curl --location --header \"Accept: application/unixref+xml\" \"http://dx.doi.org/$ref\" -o \"$outfile\" " . $redirect, $output, $return_code);
-        $return |= $return_code;
+        $ref = substr($ref, strlen('doi:'));
+        $safedoi = str_replace(')', '_', str_replace('(', '_', str_replace('/', '_', $ref)));
+        $outfile = "tmp/ref_$safedoi.xml";
+        if(!file_exists($outfile))
+        {
+            exec("curl --location --header \"Accept: application/unixref+xml\" \"http://dx.doi.org/$ref\" -o \"$outfile\" " . $redirect, $output, $return_code);
+            $return |= $return_code;
+        }
+    }
+    else if(strpos($ref, 'isbn:') === 0)
+    {
+        $ref = substr($ref, strlen('isbn:'));
+        $isbn = $ref;
+        $outfile = "tmp/ref_$isbn.xml";
+        if(file_exists($outfile))
+            continue;
+            
+        $response = file_get_contents('https://www.googleapis.com/books/v1/volumes?q=isbn:' . $isbn);
+        $results = json_decode($response);
+        if(!isset($results->items) || !count($results->items))
+        {
+            $return |= 1;
+            continue;
+        }
+        $book = $results->items[0];
+        $fp = fopen($outfile, 'w+');
```

Comment:
```bash
./basex/bin/basex -q'let $url := "https://www.googleapis.com/books/v1/volumes?q=isbn:9780199665150" return json-to-xml(fetch:text($url))'

<map xmlns="http://www.w3.org/2005/xpath-functions">
  <string key="kind">books#volumes</string>
  <number key="totalItems">1</number>
  <array key="items">
    <map>
      <string key="kind">books#volume</string>
      <string key="id">WlgLgUfZaTwC</string>
...
```

Also note you can use file_put_contents and SimpleXMLElement and generate xml properly instead of string appending.